network/endpoint.go

```diff
@@ -347,3 +347,21 @@ func GetPodNameWithoutSuffix(podName string) string {
 	logger.Info("Pod name after splitting based on", zap.Any("nameSplit", nameSplit))
 	return strings.Join(nameSplit, "-")
 }
+
+// GetEndpointInfoByIP returns an EndpointInfo with complete state when HNS Endpoint ID or HostVeth Name are missing on the CNS state.
+func (epInfo *EndpointInfo) GetEndpointInfoByIP(ipAddresses []net.IPNet, networkID string) (*EndpointInfo, error) {
+	// Call the platform implementation.
+	endpointInfo, err := epInfo.GetEndpointInfoByIPImpl(ipAddresses, networkID)
+	if err != nil {
+		return nil, err
+	}
+	return endpointInfo, nil
+}
+
+// IsEndpointStateComplete returns true if both HNSEndpointID and HostVethName are missing.
+func (epInfo *EndpointInfo) IsEndpointStateIncomplete() bool {
+	if epInfo.HNSEndpointID == "" && epInfo.IfName == "" {
```

Comment:
i guess `epInfo.Ifname` is not hostvethname. there would be field in epInfo explictly saying hostveth interface name

Reply:
We do have a HostIfName on ednpoint. DO you want me to add it to endpointInfo struct too?

Reply:
This would always return `false` for linux ? As HNSendpointId is empty for linux